mountpoint-s3-fs/src/prefetch/caching_stream.rs

```diff
@@ -378,7 +375,13 @@ where
 
 /// Creates a Part that can be streamed to the prefetcher if the given bytes
 /// are in the request range, otherwise return None.
-fn try_make_part(bytes: &ChecksummedBytes, offset: u64, object_id: &ObjectId, range: &RequestRange) -> Option<Part> {
+fn try_make_part(
+    bytes: &ChecksummedBytes,
+    offset: u64,
+    object_id: &ObjectId,
+    range: &RequestRange,
+    from_cache: bool,
```

Comment:
An enum would be preferable here so its easy to read. Something like `PartSource::Cache`, `PartSource::S3`.